js/components/ChartSummary.vue

```diff
@@ -0,0 +1,188 @@
+      <tbody>
+        <tr v-for="group in groupStats" :key="group.label">
+          <td>{{ group.label }}</td>
+          <td>{{ group.count }}</td>
+          <td>{{ group.target }}</td>
+          <td>{{ group.percentOfTarget }}</td>
+          <td>{{ group.percentOfTotal }}</td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+
+<script>
+import { copyContents, notAnsweredLabel, title } from '@/util';
+
+const messages = {
+  copyTable: 'Copy table to clipboard',
+  groupedCaption: 'Grouped Results',
+  percent: 'Percent',
+  percentOfTotal: 'Percent of Total',
+  resultsCount: 'Results Count',
+  target: 'Target',
+  totalsCaption: 'Total Number of Results',
+};
+
+/**
+ * Chart summary component. Displays counts and percentages in a tabular format.
+ */
+export default {
+  name: 'ChartSummary',
+
+  props: {
+    group: String,
+    groupData: Object,
+    totalCount: Number,
+    totalTarget: Number
+  },
+
+  data() {
+    return {
+      messages
+    };
+  },
+
+  methods: {
+    /**
+     * Calculate the percent and return a string for display.
+     * @param {Number} num - numerator
+     * @param {Number} total - denominator
+     * @return {String} percent as a string with a single decimal; ex. '32.3%'
+     */
+    pct(num, total) {
+      if (isNaN(parseFloat(num)) || isNaN(parseFloat(total)) || total === 0) {
+        return "";
+      }
+
+      const value = (num / total) * 100;
+      return `${value.toFixed(2)}%`;
+    },
+
+    /**
+     * Returns a string representation of a target for display.
+     * @param {Number} target - the target to convert
+     * @return {String} target as a string; calculated targets with fractional values are
+     *   rounded to two decimals
+     */
+    targetString(target) {
+      if (!target) {
+        return "";
+      }
+
+      return Number.isInteger(target) ? target.toString() : target.toFixed(2);
+    },
+
+    /**
+     * Generates a breakdown of statistics by group that can be rendered in a table.
+     * @param {Object} groupData - { groupName: {count: _ , target: _ }, ...}
+     * @return {Object[]} - ex.
+     * [
+     *  {label: group1Name, count: _, target: _, percentOfTarget: _, percentOfTotal: _},
+     *  ...
+     * ]
+     */
+    statistics(groupData) {
+      const groups = Object.keys(groupData);
+      groups.sort();
+
+      const total = groups.reduce((c, k) => {
+        return c + groupData[k].count;
+      }, 0);
+
+      return groups.map(group => {
+        const groupLabel = group ? group : notAnsweredLabel;
+        const data = groupData[group];
+        const { count, target } = data;
+        // Group target must exist and be non-zero to display
+        const targetDisplay = this.targetString(target); // optional form field
+        const countStr = (count !== undefined && count !== null) ? count.toString() : "";
+        // return [groupLabel, countStr, targetDisplay, this.pct(count, target), this.pct(count, total)];
```

Comment:
Is this `return` still needed?

Reply:
No, I just forgot to delete the old code. I'll remove it.